src/main/java/org/folio/linked/data/service/impl/ResourceServiceImpl.java

```diff
@@ -75,8 +81,13 @@ public Long createResource(org.folio.ld.dictionary.model.Resource modelResource)
     var mapped = resourceModelMapper.toEntity(modelResource);
     var persisted = saveMergingGraph(mapped);
     log.info("createResource [{}]\nfrom modelResource [{}]", persisted, modelResource);
-    applicationEventPublisher.publishEvent(new ResourceCreatedEvent(persisted.getId()));
-    return persisted.getId();
+    var persistedId = persisted.getId();
+    if (isAuthority(persisted)) {
+      applicationEventPublisher.publishEvent(new ResourceAuthorityCreatedEvent(persistedId));
```

Comment:
authority is also resource. That's why I think we better keep only one event: ResourceCreatedEvent.
And this logic should be done by event handler, which responsibility to distinguish which action should be done - work indexing in the Search or instance sync in the Inventory or authority indexing in the Search

Reply:
Yes. This is a good idea.